### PR TITLE
Add haptics for merge results

### PR DESCRIPTION
**Pull request by akarataev**
I've implemented a haptic handler for a successful and unsuccessful merge results.

### CODE REVIEW

**Comment by akarataev** [comment]
@Huddie, 

I tested this feature on a test repository before sending PR. 
Everything worked well.

File: Classes/Issues/Merge/GithubClient+Merge.swift [hunk 1]
```diff
@@ -18,7 +18,7 @@ extension GithubClient {
         repo: String,
         number: Int,
         type: IssueMergeType,
-        error: @escaping () -> Void
+        completionHandler: @escaping (Bool) -> Void
```
Comment (BasThomas) [inline Classes/Issues/Merge/GithubClient+Merge.swift]: Can we add a label to this tuple, like `@escaping (success: Bool) -> Void`? That makes it a bit easier to understand what the `Bool` is referring to.

File: Classes/Issues/Merge/IssueMergeSectionController.swift [hunk 2]
```diff
@@ -74,9 +74,14 @@ ListBindingSectionControllerSelectionDelegate {
             repo: model.repo,
             number: model.number,
             type: preferredMergeType,
-            error: { [weak self] in
-                self?.loading = false
-                self?.update(animated: true)
+            completionHandler: { [weak self] isSuccessMerge in
```
Comment (BasThomas) [inline Classes/Issues/Merge/IssueMergeSectionController.swift]: Super-nit: `isSuccessfulMerge`?

File: Classes/Issues/Merge/GithubClient+Merge.swift (hunk 1 above)
Comment (akarataev) [reply Classes/Issues/Merge/GithubClient+Merge.swift]: @BasThomas yes, you can.

File: Classes/Issues/Merge/IssueMergeSectionController.swift (hunk 2 above)
Comment (akarataev) [reply Classes/Issues/Merge/IssueMergeSectionController.swift]: Yes, it would be better

File: Classes/Issues/Merge/GithubClient+Merge.swift (hunk 1 above)
Comment (akarataev) [reply Classes/Issues/Merge/GithubClient+Merge.swift]: Sorry for bad english 😅
Do you want to fix it yourself or so that I fix it?

Comment (BasThomas) [reply Classes/Issues/Merge/GithubClient+Merge.swift]: If you want to fix it, go for it!

Comment (akarataev) [reply Classes/Issues/Merge/GithubClient+Merge.swift]: All is ready

File: Classes/Issues/Merge/GithubClient+Merge.swift [hunk 3]
```diff
@@ -18,7 +18,7 @@ extension GithubClient {
         repo: String,
         number: Int,
         type: IssueMergeType,
-        error: @escaping () -> Void
+        completionHandler: @escaping (_ success: Bool) -> Void
```
Comment (BasThomas) [inline Classes/Issues/Merge/GithubClient+Merge.swift]: Why do you ignore it still? That wouldn't help to understand it at the call site 😄

Comment (akarataev) [reply Classes/Issues/Merge/GithubClient+Merge.swift]: @BasThomas functional type can't have a argument label. If you ask about it.

Comment (BasThomas) [reply Classes/Issues/Merge/GithubClient+Merge.swift]: Ah.

**Pull request merged**